sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -280,6 +280,15 @@
         return new Bound<>(name, filepattern, type, schema, false);
       }
 
+      /**
+       * Configures the source to succeed if the specified {@code fileOrPatternSpec} does not match any
+       * files. The default behavior is to fail on empty input.
+       */
+      public Bound<T> withAllowedEmptyInput() {
```

Comment:
I've only added the new API on `Bound<>` PTransforms, not as a top-level `static` method to start the builder chain. I think this is sufficient because it makes sense to specify your input before specifying how validation works on it. Let me know if you think we should include the top-level static API as well.